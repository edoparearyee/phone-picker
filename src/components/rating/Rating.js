import React, { Component } from 'react';
import Star from '../star/Star';
import './Rating.css';

class Rating extends Component {
  render() {
    const stars = [0, 1, 2, 3, 4].map(i => (
      <li className="StarList--item" key={i}>
        <Star active={i < this.props.value} />
      </li>
    ));
    return <ul className="StarList">{stars}</ul>;
  }
}

export default Rating;
