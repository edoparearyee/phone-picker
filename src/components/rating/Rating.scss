.StarList {
  display: flex;
  list-style: none;
}
