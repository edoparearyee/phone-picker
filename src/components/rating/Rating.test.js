import React from 'react';
import ReactDOM from 'react-dom';
import Rating from './Rating';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Rating />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Rating has 0 active Stars', () => {
  const component = renderer.create(<Rating value={0} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Rating has 1 active Star', () => {
  const component = renderer.create(<Rating value={1} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Rating has 2 active Stars', () => {
  const component = renderer.create(<Rating value={2} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Rating has 3 active Stars', () => {
  const component = renderer.create(<Rating value={3} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Rating has 4 active Stars', () => {
  const component = renderer.create(<Rating value={4} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Rating has 5 active Stars', () => {
  const component = renderer.create(<Rating value={5} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
