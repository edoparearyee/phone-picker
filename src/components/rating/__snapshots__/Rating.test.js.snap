// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating has 0 active Stars 1`] = `
<ul
  className="StarList"
>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
</ul>
`;

exports[`Rating has 1 active Star 1`] = `
<ul
  className="StarList"
>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
</ul>
`;

exports[`Rating has 2 active Stars 1`] = `
<ul
  className="StarList"
>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
</ul>
`;

exports[`Rating has 3 active Stars 1`] = `
<ul
  className="StarList"
>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
</ul>
`;

exports[`Rating has 4 active Stars 1`] = `
<ul
  className="StarList"
>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--inactive"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
</ul>
`;

exports[`Rating has 5 active Stars 1`] = `
<ul
  className="StarList"
>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
  <li>
    <svg
      className="Star Star--active"
    >
      <polygon
        points="9.9, 1.1, 3.3, 21.78, 19.8, 8.58, 0, 8.58, 16.5, 21.78"
        style={
          Object {
            "fillRule": "nonzero",
          }
        }
      />
    </svg>
  </li>
</ul>
`;
